lib/addons/cloudwatch-logs/index.ts

```diff
@@ -0,0 +1,113 @@
+import { Construct } from "constructs";
+import * as iam from "aws-cdk-lib/aws-iam";
+import merge from "ts-deepmerge";
+import { conflictsWith, setPath } from "../../utils";
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+import { ClusterInfo, Values } from "../../spi/types";
+import { createNamespace } from "../../utils/namespace-utils";
+import { getCloudWatchLogsPolicyDocument } from "./iam-policy";
+
+/**
+ * Configuration options for the FluentBit add-on.
+ */
+export interface CloudWatchLogsAddonProps extends HelmAddOnUserProps {
+    /**
+     * Create Namespace with the provided one (will not if namespace is kube-system)
+     */
+    createNamespace?: boolean
+
+    /**
+     * Name of the service account for fluent bit.
+     */
+    serviceAccountName?: string;
+
+    /**
+     * CloudWatch Log Group Name.
+     */
+    logGroupPrefix?: string;
+
+    /**
+     * CloudWatch Log retention days
+     */
+    logRetentionDays?: number;
+}
+/**
+ * Default props for the add-on.
+ */
+const defaultProps: CloudWatchLogsAddonProps = {
+    name: 'aws-for-fluent-bit',
+    chart: 'aws-for-fluent-bit',
+    release: "blueprints-addon-aws-fluent-bit-for-cw",
+    version: '0.1.24',
+    repository: 'https://aws.github.io/eks-charts',
+    namespace: 'aws-for-fluent-bit',
+    createNamespace: true,
+    serviceAccountName: 'aws-fluent-bit-for-cw-sa',
+    logGroupPrefix: '/aws/eks/blueprints-construct-dev', 
```

Comment:
Why hardcoded prefix?

Reply:
What else is your proposal? You wanted it be passed by the user right?

Reply:
Yes,  I cannot imagine any customer who will ever want to see a log group prefx that states blueprint-construct-dev. 
I can however, imagine a prefix that is defaulted to the `eks/${clusterName}`. So let's do that. 
You can set the value of the default as "auto" and set it to the cluster name. Or customers can pass it directly.